ouroboros-network/src/Ouroboros/Network/PeerSharing.hs

```diff
@@ -104,11 +115,99 @@ peerSharingClient controlMessageSTM
         putMVar resultQueue result
         peerSharingClient controlMessageSTM psc
 
-peerSharingServer :: Monad m
-                  => (PeerSharingAmount -> m [peer])
+peerSharingServer :: ( MonadSTM m
+                     , MonadMonotonicTime m
+                     , Hashable peer
+                     , RandomGen s
+                     )
+                  => PeerSharingAPI peer s m
                   -> PeerSharingServer peer m
-peerSharingServer computePeersToShare =
+peerSharingServer psAPI =
   PeerSharingServer
-    { recvMsgShareRequest = \amount -> (,) <$> computePeersToShare amount
-                                          <*> return (peerSharingServer computePeersToShare)
+    { recvMsgShareRequest = \amount -> (,) <$> computePeerSharingPeers psAPI amount
+                                          <*> return (peerSharingServer psAPI)
     }
+
+--
+-- Utility Function
+--
+
+-- | PeerSharingAPI needed to compute the peers to be shared.
+--
+data PeerSharingAPI addr s m =
+  PeerSharingAPI { psPublicPeerSelectionStateVar :: StrictTVar m (PublicPeerSelectionState addr)
+                 , psGenVar                      :: StrictTVar m s
+                 , psReSaltAtVar                 :: StrictTVar m Time
+                 , psPolicyPeerShareStickyTime   :: !DiffTime
+                 -- ^ Amount of time between changes to the salt used to pick peers to
+                 -- gossip about.
+                 , psPolicyPeerShareMaxPeers     :: !PeerSharingAmount
+                 -- ^ Maximum number of peers to respond with in a single request
```

Comment:
Do we want these things to be configurable?

Reply:
That's a good question. I don't know? Maybe?

Reply:
`psPolicyPeerShareStickyTime` could be somewhat useful if made configurable.
A lower value would let a nodes peers be spread around more quickly. Possibly useful if one operate a peer with lots of known peers.

`psPolicyPeerShareMaxPeers` same reason, for a node with lots of know peers

Reply:
I will create an issue to make this configurable.

Reply:
https://github.com/IntersectMBO/ouroboros-network/issues/4801